src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/slurm/SlurmJobManager.groovy

```diff
@@ -138,136 +149,174 @@ class SlurmJobManager extends GridEngineBasedJobManager {
+        jobInfo.user = jobResult["UserId"]
+        jobInfo.submissionHost = jobResult["BatchHost"]
+        jobInfo.executionHosts = [jobResult["NodeList"] as String]
+        jobInfo.errorLogFile = new File(jobResult["StdErr"])
+        jobInfo.execHome = jobResult["WorkDir"]
+
+        /** Status info */
+        jobInfo.jobState = parseJobState(jobResult["JobState"])
+        jobInfo.exitCode = jobInfo.jobState == JobState.COMPLETED_SUCCESSFUL ? 0 : (jobResult["ExitCode"].split(":")[0] as Integer)
+        jobInfo.pendReason = jobResult["Reason"]
+
+        /** Resources */
+        String queue = jobResult["Partition"]
+        Duration runLimit = safelyParseColonSeparatedDuration(jobResult["TimeLimit"])
+        Duration runTime = safelyParseColonSeparatedDuration(jobResult["RunTime"])
+        jobInfo.runTime = runTime
+        BufferValue memory = safelyCastToBufferValue(jobResult["mem"])
+        Integer cores = withCaughtAndLoggedException { jobResult["cpu"] as Integer }
+        Integer nodes = withCaughtAndLoggedException { jobResult["node"]?.split("-")?.last() as Integer }
+
+        /**
+         * The Info about usedResources is not available, but to be consistent with the output expected
+         * of queryExtendedJobStateById they are both set here.
+         **/
+        jobInfo.askedResources = new ResourceSet(memory, cores, nodes, runLimit, null, queue, null)
+        jobInfo.usedResources = new ResourceSet(memory, cores, nodes, runTime, null, queue, null)
+
+        /** Timestamps */
+        jobInfo.submitTime = parseTime(jobResult["SubmitTime"])
+        jobInfo.eligibleTime = parseTime(jobResult["EligibleTime"])
+        jobInfo.startTime = parseTime(jobResult["StartTime"])
+        jobInfo.endTime = parseTime(jobResult["EndTime"])
+
+        return jobInfo
+    }
+
+    protected static Duration safelyParseColonSeparatedDuration(String value) {
         withCaughtAndLoggedException {
-            value?.length() > 7 ? parseColonSeparatedHHMMSSDuration(value[-8..-1]) : null
+            if (value.contains("-")) {
+                return Duration.ofDays(value.substring(0, value.lastIndexOf("-")).toLong()) +
+                        parseColonSeparatedHHMMSSDuration(value.substring(value.lastIndexOf("-") + 1))
+            } else {
+                return parseColonSeparatedHHMMSSDuration(value)
+            }
         }
     }
 
     /**
-     * Reads the sacct output as Json and creates GenericJobInfo objects
+     * Reads the sacct output for a single job as Json and creates GenericJobInfo objects
      * @param resultLines - Input of ExecutionResult object
      * @return map with jobid as key
      */
-    protected Map<BEJobID, GenericJobInfo> processExtendedOutputFromJson(String rawJson, Map<BEJobID, GenericJobInfo> result) {
-        if (!rawJson)
+    protected GenericJobInfo processExtendedOutputFromJson(String rawJson, GenericJobInfo result = null) {
+        if (!rawJson) {
```

Comment:
If you pull the null-value handling out (which I would prefer) you could write

```groovy
Objects.requireNonNull(rawJson)
```

here, to enforce non-null.